### DISCUSSION TITLE
How to get a menu bar using a SingleWindowWithMenuBar?

### DISCUSSION THREAD

**Discussion by manuel-koch** [AllenDang/giu#617] [Q&A] [answered]
I'm trying to add a menubar to my window, by adapting the [example](https://github.com/AllenDang/giu/blob/master/examples/multiwindow/multiwindow.go).

But it doesn't work, what is my error ?

E.g. within my `loop()`, using the following results in a main menu for a `Window`:
```go
giu.MainMenuBar().Layout(
    giu.Menu("Help").Layout(
        giu.MenuItem("About").OnClick(func() { fmt.Println("ABOUT") }),
    ),
).Build()

giu.Window("Window 1").Pos(10, 30).Size(200, 100).Layout(
    giu.Label("I'm a label in window 1"),
    giu.Button("Show Window 2").OnClick(func() { fmt.Println("CLICKED") }),
)
```
![menu-bar-window](https://user-images.githubusercontent.com/8177285/205023020-1e7c75dc-de7f-4afe-aaa3-bb58be3b0b9f.png)

**BUT** e.g. adapting the example to use a `SingleWindowWithMenuBar` instead results in no main menu, just an empty colored area at the top of the window that looks like it should be a menu !?

```go
giu.MainMenuBar().Layout(
    giu.Menu("Help").Layout(
        giu.MenuItem("About").OnClick(func() { fmt.Println("ABOUT") }),
    ),
).Build()

giu.SingleWindowWithMenuBar().Layout(
  ....
)
```
![empty-menu-bar](https://user-images.githubusercontent.com/8177285/205023066-6987e764-5330-41a0-8128-c8141a05aafa.png)

and equally empty main menu area with the `MainMenuBar` within the `SingleWindowWithMenuBar`

```go
giu.SingleWindowWithMenuBar().Layout(
  giu.MainMenuBar().Layout(
      giu.Menu("Help").Layout(
          giu.MenuItem("About").OnClick(func() { fmt.Println("ABOUT") }),
      ),
  ),
)
```

slightly changing the code results in a broken main menu, i.e. there is still this empty area but the menu item appears somewhere below ( but not within the empty area ):

```go
giu.SingleWindowWithMenuBar().Layout(
      giu.Menu("Help").Layout(
          giu.MenuItem("About").OnClick(func() { fmt.Println("ABOUT") }),
      ),
)
```
![broken-menu-bar](https://user-images.githubusercontent.com/8177285/205023128-547787ff-b886-4e11-8e9a-4218713e7b73.png)

**Comment by gucio321** [comment, marked as answer]
I remember i did it working on my https://github.com/thegreaterheptavirate/constitutor
It is in pkg/app
I can post you a exact code a bit later

**Reply by manuel-koch** [reply]
Thanks for your hint. Looking at your code I spotted a slight difference that did the trick.
One has to use `MenuBar` within a `SingleWindowWithMenuBar` instead of a `MainMenuBar ` !

The following works as expected :

```go
giu.SingleWindowWithMenuBar().Layout(
    giu.MenuBar().Layout(
        giu.Menu("Help").Layout(
            giu.MenuItem("About").OnClick(func() { fmt.Printf("ABOUT") }),
        ),
    ),
    ....
)
```
